### PR TITLE
Fix layout indication in various cases

### PR DESCRIPTION
**Pull request by im-0**

### CODE REVIEW

File: seat.c [hunk 1]
```diff
@@ -82,7 +82,7 @@ static void keyboard_modifiers(void *data, struct wl_keyboard *wl_keyboard,
 	xkb_state_update_mask(state->xkb.state,
 		mods_depressed, mods_latched, mods_locked, 0, 0, group);
 	int caps_lock = xkb_state_mod_name_is_active(state->xkb.state,
-		XKB_MOD_NAME_CAPS, XKB_STATE_MODS_LOCKED);
+		XKB_MOD_NAME_CAPS, XKB_STATE_MODS_EFFECTIVE);
```
Comment (emersion) [inline seat.c]: Is there an issue with LOCKED?

Comment (im-0) [reply seat.c]: No issue. I just thought that EFFECTIVE would be more logical here.

I can remove this commit if you think that this is wrong.

File: render.c [hunk 2]
```diff
@@ -168,16 +168,7 @@ void render_frame(struct swaylock_surface *surface) {
 			xkb_layout_index_t num_layout = xkb_keymap_num_layouts(state->xkb.keymap);
 			if (!state->args.hide_keyboard_layout &&
 					(state->args.show_keyboard_layout || num_layout > 1)) {
```
Comment (im-0) [inline render.c]: Also, maybe I should add `&& state->xkb.layout_set` here?

Is it possible that this code will be executed before current keyboard layout is set by keyboard_modifiers()?

**Comment by im-0** [comment]
Any news?

I'm using these patches without any problems, by the way.

**Comment by im-0** [comment]
Seems not needed for newer version of sway/swaylock

**Pull request closed without merging**